### ISSUE TITLE
SSL redirect to non SSL site

### ISSUE THREAD

**Issue by Talwoasc** [quadshot/opentickets-community#2]
Hi,

Great plugin but I've noticed that when this plugin is used with "Wordpress HTTPS" it causes any https page request to be 301 redirected to the http version of the site. When either "Wordpress HTTPS" or "OpenTickets" is not activated the redirects do not occur, it seems to be a combination of the two causing the problems. Even when all the options in "Wordpress HTTPS" are disabled and I disable the force SSL checkout in "WooCommerce" the same thing happens.
I've replicated this on a blank install to check it wasn't something else I had missed. Had a look through your code but I can't seem to spot anything that might cause this but obviously I'm not very familiar with your code!

Thanks,
Tal

**Comment by michaelcremean** [comment]
Hello,

Thanks for the thorough explanation. We were able to replicate this on a site using the same combination of plugins. 

Our lead developer is working on a solution for this 'catch 22' scenario. We'll be updating the core OT when it is, so you can upgrade and solve this problem.

It actually has to do with forcing https in the admin itself, for example with iThemes security settings, causing a conflict. 

If you aren't already, if you subscribe to the mailing list on OpenTickets.com we'll email when we do major updates like this.

**Comment by loushou** [comment]
Hey @Talwoasc ,

After some debugging and some research, it turns out that this is actually a conflict with WooCommerce and WordPress HTTPS, and a known one. Check http://docs.woothemes.com/document/ssl-and-https/#section-5 under 'Redirect Loops'. Also, there are several forum posts about it, for instance https://wordpress.org/support/topic/redirect-loop-error-with-woocommerce .

The basic problem is this. Both WordPress HTTPS and WooCommerce use their own method of determining if pages need to be https or not. When both of them are not set to the exact same settings for a given page, they will continue to try to 'correct' the other one. To make matters worse, 'disabling' all options in the WordPress HTTPS plugin actually does not prevent the plugin from doing stuff. Disabling all the options actually tells the plugin to 'unHTTPS' all urls. Thus, when you have the plugin active and 'disable' all it's settings, while having WooCommerce 'force ssl checkout' enabled, WooCommerce says 'this page should be SSL' and WordPress HTTPS says 'no... this page should be non-SSL'. They keep going back and forward, and eventually your browser says 'ENOUGH! neither of you win'.

As for the case you report of '...I disable the force SSL checkout in "WooCommerce" the same thing happens...', I can only replicate this in a browser that does not have all cache disabled. Basically, modern browsers (namely Chrome) not only cache pages and js and css, but also cache seemingly mundane things like DNS, certain headers, and redirects. So if you continued to experience this problem even after disabling WooCommerce 'force SSL checkout', then you were getting hit by this browser 'feature'. You can prove this to yourself by performing the following steps:

- 'disable' all WordPress HTTPS options, but keep the plugin active
- turn ON the 'force ssl checkout' option in WooCommerce
- on the frontend of your site, attempt a checkout, noticing that there is a redirect loop
- go back to the admin, and turn OFF the WooCommerce 'force ssl checkout' setting
- attempt to refresh the checkout page, noticing you still have a redirect loop
- clear your browser cache and _completely_ restart the browser (do not try to circumvent this step by using incognito, because it uses the same cached redirects)
- try to goto the checkout a final time, and notice that it works now

In conclusion, there are a few workarounds for this conflict, a couple of which are mentioned in that example forum post above. WooCommerce itself recommends that you do not use WordPress HTTPS because of this very redirect problem. My personal recommendation is to completely disable the WooCommerce 'force ssl checkout' option, and use WordPress HTTPS to specifically require ssl on the checkout page, and all it's child pages. This will prevent the ongoing conflict between WooCommerce and WordPress HTTPS from even being a factor.

Hope this helps, and thanks for reporting any potential issue,
Loushou

**Comment by Talwoasc** [comment]
Hi @loushou,

Thank you so much for your detailed reply! That is very helpful (sorry I didn't do better research myself). I will implement your suggestions and fingers crossed it all works.

Tal

**Comment by loushou** [comment]
@Talwoasc - No Problem at all. Glad I was able to track it down and provide information on the problem. Hopefully this will help others as well.

Loushou
